test/unit/io/structure_file/format_vienna_test.cpp

```diff
@@ -118,22 +110,20 @@ struct read : public ::testing::Test
     {
         std::stringstream istream{input};
 
-        for (size_t idx = 0ul; idx < expected_seq.size(); ++idx)
+        auto field_set = fields<field::ID, field::SEQ, field::BPP, field::STRUCTURE, field::ENERGY>{};
+        structure_file_input fin{istream, format_vienna{}, field_set};
+        fin.options = options;
+
+        auto it = fin.begin();
+        for (size_t idx = 0ul; idx < expected_seq.size(); ++idx, ++it)
         {
-            id.clear();
-            seq.clear();
-            structure.clear();
-            bpp.clear();
-            energy = std::nullopt;
-
-            EXPECT_NO_THROW(( format.read(istream, options, seq, id, bpp, structure, energy, ig, ig, ig, ig) ));
-
-            EXPECT_EQ(check_energy, energy.has_value());
-            if (check_seq)       { EXPECT_TRUE(( std::ranges::equal(seq, expected_seq[idx]) )); }
-            if (check_id)        { EXPECT_TRUE(( std::ranges::equal(id, expected_id[idx]) )); }
-            if (check_structure) { EXPECT_TRUE(( std::ranges::equal(structure, expected_structure[idx]) )); }
-            if (check_structure) { bpp_test(expected_interactions[idx]); }
-            if (check_energy)    { EXPECT_DOUBLE_EQ(*energy, expected_energy[idx]); }
+            EXPECT_EQ(check_energy, get<field::ENERGY>(*it).has_value());
+            if (check_seq)       { EXPECT_TRUE((std::ranges::equal(get<field::SEQ>(*it), expected_seq[idx]))); }
```

Comment:
missing include for std::ranges::equal?

Reply:
it is defined in `seqan3/std/algorithm` which is included :)